src/main/java/org/ossgang/commons/observables/Transition.java

```diff
@@ -0,0 +1,65 @@
+package org.ossgang.commons.observables;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents a transition from an old value of an observable value to a new one. This is used in accumulate operations
+ * of
+ * properties.
+ */
+public class Transition<T> {
+
+    private final T oldValue;
+    private final T newValue;
+
+    private Transition(T oldValue, T newValue) {
+        this.oldValue = oldValue; /* The old value is allowed to be null */
+        this.newValue = requireNonNull(newValue, "new value must not be null");
```

Comment:
I would not enforce null or non-null here - I think ideally the knowledge that the previous value can be null (in the very special case of an uninitialized observable) but the updated value can should be in `DispatchingObservableValue` and nowhere else.

Reply:
Ok. Sounds reasonable.

Reply:
Done